be/src/exec/tablet_sink.cpp

```diff
@@ -1149,9 +1262,39 @@ Status OlapTableSink::send_chunk(RuntimeState* state, Chunk* chunk) {
         {
             uint32_t num_rows_after_validate = SIMD::count_nonzero(_validate_selection);
             int invalid_row_index = 0;
-            RETURN_IF_ERROR(_vectorized_partition->find_tablets(chunk, &_partitions, &_tablet_indexes,
-                                                                &_validate_selection, &invalid_row_index));
 
+            if (_enable_automatic_partition) {
+                _partition_not_exist_row_values.clear();
+                // only support single column partition now
+                _partition_not_exist_row_values.resize(1);
+                RETURN_IF_ERROR(_vectorized_partition->find_tablets(chunk, &_partitions, &_tablet_indexes,
+                                                                    &_validate_selection, &invalid_row_index, _txn_id,
+                                                                    &_partition_not_exist_row_values));
+
+                if (!_partition_not_exist_row_values[0].empty()) {
+                    _is_automatic_partition_running.store(true, std::memory_order_release);
+                    RETURN_IF_ERROR(_automatic_partition_token->submit_func([this] {
```

Comment:
If the ingestion failed, the partition will be rollbacked or not?

Reply:
not